services/yellow/src/contracts/Models/PaymentAuthorizationToken.cs

```diff
@@ -4,24 +4,24 @@ namespace artiso.AdsdHotel.Yellow.Contracts.Models
 {
     public class PaymentAuthorizationToken
     {
-        public PaymentAuthorizationToken()
+        public PaymentAuthorizationToken(Guid token)
         {
-            
+            Token = token.ToString();
         }
 
         public PaymentAuthorizationToken(TimeSpan duration)
         {
             Token = Guid.NewGuid().ToString();
             CreatedAt = DateTime.Now;
-            ExpirationDate = CreatedAt.Value.Add(duration);
+            ExpirationDate = CreatedAt.Add(duration);
             Active = true;
         }
 
         public string? Token { get; init; }

```

Comment:
Very confusing if it is a constructor parameter and a property initializer. Stick to one.
```suggestion
        public string? Token { get; }

```